.github/workflows/publish_plugin_to_hub.yaml

```
@@ -0,0 +1,120 @@
+        id: semver_parser
+        with:
+          input_string: ${{steps.split.outputs.plugin_version}}
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: actions/github-script@v6
+        id: set-result
+        env:
+          PLUGIN_DIR: ${{steps.split.outputs.plugin_dir}}
+        with:
+          script: |
+            const fs = require('fs').promises;
+            const path = require('path');
+            const pluginFiles = await fs.readdir(process.env.PLUGIN_DIR);
+            if (pluginFiles.includes('Dockerfile')) {
+              return 'docker';
+            }
+            if (pluginFiles.includes('.goreleaser.yaml')) {
+              return 'go';
+            }
+          result-encoding: string
+  publish-plugin-to-hub:
+    timeout-minutes: 60
+    runs-on: large-ubuntu-monorepo
+    needs: prepare
+    if: needs.prepare.outputs.plugin_releaser == 'go'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-1.21.1-publish-plugin-to-hub-cache
+          restore-keys: |
+            ${{ runner.os }}-go-1.21.1-publish-plugin-to-hub-cache
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: ${{ needs.prepare.outputs.plugin_dir }}/go.mod
+          cache: false
+
+      - name: Use Node.js LTS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+
+      - name: Install shell escape
+        run: |
+          npm install shell-escape@0.2.0
+
+      - name: Get Release Notes
+        id: release-notes
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          script: |
+            const shellescape = require('shell-escape');
+            const { data } = await github.rest.repos.getReleaseByTag({
+              owner: "cloudquery",
+              repo: context.repo.repo,
+              tag: context.ref.replace('refs/tags/', ''),
+            });
+            return shellescape([data.body]);
+
+      - name: Run package command
+        working-directory: ${{ needs.prepare.outputs.plugin_dir }}
+        run: |
+          go run main.go package --docs-dir docs -m ${{ steps.release-notes.outputs.result }} ${{ needs.prepare.outputs.plugin_version }} .
+
+      - name: Setup CloudQuery
+        uses: cloudquery/setup-cloudquery@v3
+        with:
+          version: v3.26.1
+
+      - name: Publish plugin to hub
+        working-directory: ${{ needs.prepare.outputs.plugin_dir }}
+        run: |
+          mkdir -p ~/.local/share/cloudquery
+          echo ${{ secrets.CQ_CI_CLOUDQUERY_HUB_TOKEN }} > ~/.local/share/cloudquery/token
+          cloudquery plugin publish scaleway/${{ needs.prepare.outputs.plugin_name }} --finalize
```

```suggestion
          cloudquery plugin publish --finalize
```

See https://github.com/cloudquery/cloudquery/pull/15079